homeassistant/components/efergy/sensor.py

```diff
@@ -93,37 +88,35 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType = None,
 ) -> None:
     """Set up the Efergy sensor."""
-    app_token = config.get(CONF_APPTOKEN)
-    utc_offset = str(config.get(CONF_UTC_OFFSET))
+    api = Efergy(
+        config.get(CONF_APPTOKEN),
+        async_get_clientsession(hass),
+        utc_offset=str(config.get(CONF_UTC_OFFSET)),
+    )
 
     dev = []
     for variable in config[CONF_MONITORED_VARIABLES]:
         if variable[CONF_TYPE] == CONF_CURRENT_VALUES:
-            url_string = f"{_RESOURCE}getCurrentValuesSummary?token={app_token}"
-            response = requests.get(url_string, timeout=10)
-            for sensor in response.json():
-                sid = sensor["sid"]
+            for sensor in await api.get_sids():
```

Comment:
Please move the api call first as the outermost loop. We only need one api call.